src/test/java/org/broadinstitute/consent/http/service/dao/DatasetServiceDAOTest.java

```diff
@@ -729,6 +730,76 @@ void testDeleteStudy() throws Exception {
 
   }
 
+  @Test
+  void testExecuteUpdateDatasetWithNullName() throws Exception {
+    // This creates a study with a single dataset:
+    Study study = createStudy(List.of());
+    List<Dataset> datasets = datasetDAO.findDatasetsByIdList(study.getDatasetIds().stream().toList());
+    Dataset dataset = datasets.get(0);
+    jdbi.useHandle(handle -> serviceDAO.executeUpdateDatasetWithFiles(
+        handle,
+        dataset.getDatasetId(),
+        null,
+        study.getCreateUserId(),
+        dataset.getDacId(),
+        List.of(),
+        List.of(),
+        false)
+    );
+    Dataset updatedDataset = datasetDAO.findDatasetById(dataset.getDatasetId());
+    assertNotNull(updatedDataset);
+    assertNotNull(updatedDataset.getName());
```

Comment:
In this case I would omit the first `assertNotNull` since it will throw a NPE if it's null and the test will fail anyway.